sql/catalyst/src/main/java/org/apache/spark/sql/connector/util/V2ExpressionSQLBuilder.java

```diff
@@ -228,4 +241,37 @@ protected String visitSQLFunction(String funcName, String[] inputs) {
   protected String visitUnexpectedExpr(Expression expr) throws IllegalArgumentException {
     throw new IllegalArgumentException("Unexpected V2 expression: " + expr);
   }
+
+  protected String visitOverlay(String funcName, String[] inputs) {
```

Comment:
can we add some `if-else` in `visitSQLFunction` to deal with these special syntax? Then we don't need to keep adding new functions to this class.

Reply:
Maybe we can remove the `funcName` parameter.